django/core/files/base.py

```diff
@@ -149,6 +149,12 @@ def open(self, mode=None):
     def close(self):
         pass
 
+    def write(self, data):
+        """ Write data and get size from underlying file """
+        result = self.file.write(data)
+        self._size = self._get_size_from_underlying_file()
```

Comment:
I think invalidating the `size` cache would make more sense than re-computing the size on every single write.

```python
self.__dict__.pop('_size', None)
```

Reply:
@charettes yes, you are right. Invalidating is more better. Thanks for quick review!